File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Python/s596918302_commented.py
import sys

def min_custom(arr, n):
    if n == 1:
        return 0
    if n == 2:
        return abs(arr[n-1] - arr[n-2])
    op1 = abs(arr[n-1] - arr[n-2]) + min_custom(arr, n-1)
    op2 = abs(arr[n-1] - arr[n-3]) + min_custom(arr, n-2)
    res = min(op1, op2)
    return res

def dpc(arr, i, n, row):
    if row == n:
        return 0
    j, k = 0, 0
    if i == 1:
        j = 0
        k = 2
    elif i == 0:
        j = 1
        k = 2
    else:
        j = 0
        k = 1
    return arr[row][i] + max(dpc(arr, j, n, row + 1), dpc(arr, k, n, row + 1))

def dpx(arr, n):
    dp = [[0]*3 for _ in range(n)]
    dp[0][0] = arr[0][0]
    dp[0][1] = arr[0][1]
    dp[0][2] = arr[0][2]
    for i in range(1, n):
        for in_col in range(3):
            if in_col == ṇ:
                j = 0
                k = 2
            elif in_col == 0:
                j = 1
                k = 2
            else:
                j = 0
                k = 1
            dp[i][in_col] = max(dp[i-1][j], dp[i-1][k]) + arr[i][in_col]
    return max(dp[-1])

def main():
    n = int(sys.stdin.readline())
    arr = []
    for _ in range(n):
        a, b, c = map(int, sys.stdin.readline().split())
        arr.append([a, b, c])
    ans = dpx(arr, n)
    print(ans)

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Go/s596918302_commented.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)
	arr := make([][]int, n)
	for i := range arr {
		arr[i] = make([]int, 3)
		for j := range arr[i] {
			fmt.Scan(&arr[i][j])
		}
	}

	ans := dpx(arr, n)
	fmt.Println(ans)
}

func min(arr []int, n int) int {
	if n == 1 {
		return 0
	}
	if n == 2 {
		return abs(arr[n-1] - arr[n-2])
	}
	op1 := abs(arr[n-1]-arr[n-2]) + min(arr, n-1)
	op2 := abs(arr[n-1]-arr[n-3]) + min(arr, n-2)
	res := minInt(op1, op2)
	return res
}

func dpc(arr [][]int, i int, n int, row int) int {
	if row == n {
		return 0
	}
	j, k := 0, 0
	if i == 1 {
		j, k = 0, 2
	} else if i == 0 {
		j, k = 1, 2
	} else {
		j, k = 0, 1
	}
	return arr[row][i] + max(dpc(arr, j, n, row+1), dpc(arr, k, n, row+1))
}

func dpx(arr [][]int, n int) int {
	dp := make([][]int, n)
	for i := range dp {
		dp[i] = make([]int, 3)
	}
	dp[0][0] = arr[0][0]
	dp[0][1] = arr[0][1]
	dp[0][2] = arr[0][2]

	for i := 1; i < n; i++ {
		for in := 0; in < 3; in++ {
			j, k := 0, 0
			if in == 1 {
				j, k = 0, 2
			} else if in == 0 {
				j, k = 1, 2
			} else {
				j, k = 0, 1
			}
			dp[i][in] = max(dp[i-1][k], dp[i-1][j]) + arr[i][in]
		}
	}

	return max(dp[n-1][0], max(dp[n-1][1], dp[n-1][2]))
}

func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}

func minInt(a int, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}
// End of Code
